### PR TITLE
Allow using environment variables instead of configuration.yml

### PR DESCRIPTION
**Pull request by mfrister**
Environment variables are a good idea for automated deployments, e.g. they can't accidentally be checked in and are a operating system concept working everywhere. See here for more reasons: http://12factor.net/config

This also documents the configuration, renames Redmine::Configuration to OpenProject::Configuration and adds a spec.

### CODE REVIEW

File: spec/lib/open_project/configuration_spec.rb [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+#
+# See doc/COPYRIGHT.rdoc for more details.
+#++
+
+require 'spec_helper'
+
+describe OpenProject::Configuration do
+  # let(:config) { OpenProject::Configuration }
+  describe 'an empty configuration' do
+    let(:config) { load_conf('empty.yml', 'test') }
```
Comment (hschink) [inline spec/lib/open_project/configuration_spec.rb]: I assume that you're not interested in testing the file loading/parsing capabilities of Rails? Thus, I propose that you either mock the file access or that you separate the file loading from the other aspects.

**Comment by hschink** [comment]
I tested the code with `AUTOLOGIN_COOKIE_NAME` and it works! :+1: But which test tests that?

**Comment by mfrister** [comment]
Restructured the Configuration module a bit to allow testing without the YAML files. Removed them and added a new test for the environment variables.

**Comment by hschink** [comment]
Very nice! :cherry_blossom: Final question: Do you need to clean up the `ENV` tests if you use

``` ruby
ENV.stub(:[]).with('SOMESETTING').and_return('bar')
```

?

**Comment by mfrister** [comment]
No, but the test also fails:

```
ENV received :[] with unexpected arguments
         expected: ("SOMESETTING")
              got: ("ANSICON")
```

Let's merge :dancer:

**Pull request merged**